_posts/en/newsletters/2019-09-18-newsletter.md

```diff
@@ -0,0 +1,312 @@
+---
+title: 'Bitcoin Optech Newsletter #64'
+permalink: /en/newsletters/2019/09/18/
+name: 2019-09-18-newsletter
+slug: 2019-09-18-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes an idea for providing SPV fraud proofs
+without changing Bitcoin's consensus rules and summarizes several talks
+from the Bitcoin Edge Dev++ training sessions and Scaling Bitcoin
+conference held last week in Tel Aviv.  Also included is our regular
+section on notable changes to popular Bitcoin infrastructure projects.
+
+{% comment %}<!-- include references.md below the fold but above any Jekyll/Liquid variables-->{% endcomment %}
+{% include references.md %}
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **SPV fraud proofs without a soft fork:** Ruben Somsen started a
```

Comment:
1. I think you answered this yourself.
2. Worst case would be the SPV client would need to download one commitment for every block in the block chain, e.g. 32 MB for 32-byte commitments on a chain with a million blocks.  Practically, I'd expect SPV clients to ship with checkpoints (many do already) that wouldn't believe a node that claimed a fork further back than a few months, or about 1 MB of commitments.  (Even more practically, though perhaps slightly risky, an SPV client may refuse to believe in forks over a week long, or about 32 kB of commitments.)  Clients can detect stalling and disconnect the offending node; Bitcoin users should be used to waiting an hour or more for ~6 confirmations, so this process doesn't need to be be super fast.

3. It's worse than storing commitments (those are just a small hash), I think they need to store (1) the UTXOs being spent in a block (i.e. undo data), (2) the utreexo partial merkle branches connecting the UTXOs to the utreexo merkle root, (3) the block itself.  My very rough guess is that this would require about double the disk space of a current archival full node.  That said, I don't see any complexity to storing and distributing this information (like BIP158 filters, it should be compute-once and store-on-disk) and if there's an expectation that SPV clients won't believe in forks more than *n* blocks old, then a node serving these fraud proofs only needs to store *n* blocks worth of this data.  All that said, this was my main concern---that too few full node operators will care to generate and serve this data, making it easy to sybil SPV clients needing that data and so making the "one honest peer" assumption even easier to violate than it is now.

4. This is explicitly mentioned in the newsletter description.  But, yeah, I don't expect SPV clients to actually do that (though some developers \*cough\*neutrino\*cough\* have been claiming that they'll do stuff like that to make SPV secure in the future).

In general, I was pretty dubious about this proposal myself and my initial response was to draft an email for the mailing list describing its flaws.  However, each flaw I thought I found was one I also found a reasonable solution for, even if I think some of them are unlikely to be implemented.  So I wrote this description because SPV fraud proofs under reasonable assumptions without a soft fork seems cool.  However, if after reading this too-long comment reply, you still think the proposal is too immature, I'm happy to drop it.  This week's newsletter is long enough already.

Reply:
Note: I just replied to Somsen on list with several of these points (plus others) so that he's aware of the possible criticism.